Document AzureEP 1.16 (#17289)
microsoft/onnxruntime: docs/execution-providers/Azure-ExecutionProvider.md

Azure Execution Provider is in preview stage, and all API(s) and usage are subject to change.

> Till onnxruntime 1.16, three operators are available:

Reviewer: 'Till 1.16'  what about what you can do now?   Seems an odd opening

Reviewer 2: How about the following? 

Starting 1.16 release AzureEP supports 3 new models hosted on both Azure and OpenAI using pluggable custom ops. Each custom op targets a specific model and can be used to compose model configurations that support various scenarios. As an example, we have written samples to show configurations that support If/Then/And scenarios. More info about these can be found here: https://github.com/microsoft/onnxruntime-inference-examples/tree/main/python/AzureEP.

Read more about these custom ops and the models they support here:
- [OpenAIAudioToText](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#openaiaudiototext)
- [AzureTextToText](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#azuretexttotext)
- [AzureTritonInvoker](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#azuretritoninvoker)

### Asynchronous Run API
Moreover, we've also added an [asynchronous Run API](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L4341) to make inferences asynchronously. This will useful when inferencing models that are hosted on remote services like Azure and OpenAI.

Author: "As an example, we have written samples to show configurations that support If/Then/And scenarios. More info about these can be found here: https://github.com/microsoft/onnxruntime-inference-examples/tree/main/python/AzureEP." 
Did this part in Usage under the 2nd mode.

"Moreover, we've also added an [asynchronous Run API](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L4341) to make inferences asynchronously. This will useful when inferencing models that are hosted on remote services like Azure and OpenAI."

Did this in Usage, under the 1st mode.

- [OpenAIAudioToText](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#openaiaudiototext)
- [AzureTextToText](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#azuretexttotext)
- [AzureTritonInvoker](https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/custom_ops.md#azuretritoninvoker)

Each of above operators could be used to compose a proxy model that talks to a supported remote endpoint.

## Contents
{: .no_toc }

* TOC placeholder
{:toc}

## Install

> Azure Execution Provider releases with default onnxruntime python and nuget packages.

Reviewer 2: ```suggestion
Azure Execution Provider is shipped by default in both python and nuget packages.
```

Author: Done.

## Requirements

> For onnxruntime 1.16, all Azure Execution Provider operators are released with onnxruntime extension (>=v0.9.0) python and nuget packages.

Reviewer 2: ```suggestion
Starting onnxruntime 1.16, all Azure Execution Provider operators are released with onnxruntime extension (>=v0.9.0) python and nuget packages.
```

Author: Done.

> Please ensure the installation of correct onnxruntime extension packages beforehand.

Reviewer 2: ```suggestion
Please ensure the installation of correct onnxruntime extension package before using AzureEP.
```

Author: Done.

{:toc}

## Install
Since 1.16, Azure Execution Provider is shipped by default in both python and nuget packages.

## Requirements
Since 1.16, all Azure Execution Provider operators are shipped with [onnxruntime-extensions](https://github.com/microsoft/onnxruntime-extensions) (>=v0.9.0) python and nuget packages. Please ensure the installation of correct onnxruntime-extension packages before using Azure Execution Provider.

## Build

For build instructions, please see the [BUILD page](../build/eps.md#azure).

## Usage

### Edge and azure side by side

> In this mode, there are two models running simulaneouly. The azure model runs asynchronously by [RunAsync](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L4341) API, which is also available through [python](https://github.com/microsoft/onnxruntime/blob/873ef8b8f0b09b49c0a7b7e2f03f3639d7418c22/onnxruntime/python/onnxruntime_pybind_state.cc#L1759) and [csharp](https://github.com/microsoft/onnxruntime/blob/873ef8b8f0b09b49c0a7b7e2f03f3639d7418c22/csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs#L1147).

Reviewer: spelling:  simultaneously
